local.c

```diff
@@ -1484,13 +1484,21 @@ local_create_buffer(const struct iio_device *dev, unsigned int idx,
 	const struct iio_channel *chn;
 	int err, cancel_fd, fd;
 	unsigned int i;
+	const char *use_cma_for_device;
 
 	pdata = zalloc(sizeof(*pdata));
 	if (!pdata)
 		return iio_ptr(-ENOMEM);
 
 	pdata->dev = dev;
 
+	pdata->dmabuf_alloc_type = DMABUF_ALLOC_SC;
+	use_cma_for_device = iio_getenv("CMA_FOR_DEVICE");
+	if (use_cma_for_device) {
+		if (strstr(dev->name, use_cma_for_device))
+			pdata->dmabuf_alloc_type = DMABUF_ALLOC_CMA;
+	}
```

Comment:
This is now just a draft so I guess we can use the general ideal for testing purposes. I would be interested to understand if the underlying kernel code works (as-is) with the CMA dma-heap (It should but still...). 

Having said the above, I do not think this is the proper way... I'm afraid we need a dedicated API for choosing the heap. The reason is that this env variable is selecting devices while this setting should be per buffer. Note that the kernel supports multiple buffers (which can be DMA) per device and while very unlikely, it should be perfectly fine to have a buffer connected to a DMA controller doing CMA and the other doing SG.

Also, this only assumes two kind of heaps while there's nothing stopping us to add more in the future. This approach won't scale that nicely.

Reply:
I had a feeling that these changes will ask for an API change, but I didn't know about the possibility of having multiple buffers.
I have tested this but open() fails on my setup (zc706 with ADI image). The `errno` gets set to ENOENT even though running this shows:
```
analog@analog:~ $ dmesg | grep -i cma
[    0.000000] cma: Reserved 128 MiB at 0x38000000
[    0.000000] Memory: 881944K/1048576K available (12288K kernel code, 818K rwdata, 10376K rodata, 1024K init, 470K bss, 35560K reserved, 131072K cma-reserved, 131072K highmem)
```

I will be working on defining a proper API but in the meantime, as you said, this can be used for testing things.
Thanks @nunojsa

Reply:
> I have tested this but open() fails on my setup (zc706 with ADI image). The errno gets set to ENOENT even though running this shows:

Likely because that module is not being compiled atm